writeup.md

```diff
@@ -0,0 +1,85 @@
+        - A line chart effectively represents trends over time
+        - Dots on the lines allow for better granular insights at each season
+        - Dual lines allow comparison between general shot efficiency and defensive contest rate
+2. Shot Efficiency vs Defensive Pressure
+    - Encoding
+        - X-axis: Shot Efficiency (Quantitative)
+        - Y-axis: Contest Percentage (Quantitative)
+        - Circle Size: Consistent
+    - Rationale
+        - A scatter plot is ideal for analyzing correlations
+        - This chart allows users to understand if contested shots significantly impact efficiency
+3. Shot Hotspots Across the Court
+    - Encoding
+        - X-axis: Court coordinates of shots (Location)
+        - Y-axis: Court coordinates of shots (Location)
+        - Color: Red to Blue (Indicating shooting accuracy)
+    - Rationale
+        - Hexbin heatmaps efficiently summarize thousands of data points.
+        - The color gradient provides an intuitive sense of high-efficiency zones (blue) vs low-efficiency zones (red).
+        - The court outline preserves spatial awareness.
+
+#### Interaction Techniques
+1. Search & Filter by Player / Team  
+Users can quickly explore individual player tendencies or team shooting styles
+2. Smooth Transitions for Data Updates  
+This helps maintain visual continuity, preventing sudden jumps in the data
+3. Hover Interactions for Data Tooltips  
+This provides detailed information without cluttering the charts
+
+#### Alternatives Considered
+1. Bar Charts for Trends
+    - Line charts showcase trends better over time
+2. Shot Location as Scatter Dots
+    - Too much clutter with raw shot data, hexbin aggregation makes trends more interpretable
+3. Static Filters Instead of Search
+    - Searchable player/team selector improves usability over dropdowns
+
+#### References
+Data Sources
+- https://www.nba.com/stats
+- https://github.com/swar/nba_api
+
+#### Development Process
+The development of NBA Shot Insights: Trends, Efficiency & Court Hotspots was a collaborative effort between Kun and James, with work divided based on data preparation, initial implementation, and final refinements. We collaborated asynchronously to work as both of us had time. 
+
+Team Contributions  
+- Kun:
+    - Collected and cleaned the NBA shooting data.
+    - Conducted data preprocessing to ensure accuracy.
+    - Implemented initial versions of the three charts.
+    - Implemented initial version of the search bar.
+    - Implemented interactive features such as hover
+- James:
+    - Refactored the initial chart implementations and finalized the visualizations.
+    - Polished interactive features, including:
+        - Search & filter functionality for players and teams.
+        - Smooth transitions for updated data.
+    - Designed styling, layout, and UI polish for the final visualization.
+
+The project took approximately () hours in total
+- Data collection & cleaning - (Kun to add estimate)
```

Comment:
@kuqin12 can you add time estimates here and update the total hours

Reply:
```suggestion
- Data collection & cleaning - 4 hours
```